File: server/HabitTracker.Api/DTOs/DashboardHabitResponse.cs
using HabitTracker.Api.Domain.Enums;

namespace HabitTracker.Api.DTOs;

public sealed class DashboardHabitResponse
{
    public Guid Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public HabitCategory Category { get; set; }

    public HabitFrequencyType FrequencyType { get; set; }

    public int TargetCount { get; set; }

    public HabitDifficulty Difficulty { get; set; }

    public IReadOnlyList<HabitAttributeRewardResponse>
        AttributeRewards
    { get; set; } = [];

    public bool IsCompletedToday { get; set; }

    public int CurrentStreak { get; set; }

    public int LongestStreak { get; set; }
}

File: client/src/types/DashboardHabitResponse.ts
import type { HabitAttributeRewardResponse } from './HabitAttributeRewardResponse'
import type { HabitCategory } from './HabitCategory'
import type { HabitDifficulty } from './HabitDifficulty'
import type { HabitFrequencyType } from './HabitFrequencyType'

export type DashboardHabitResponse = {
  id: string
  name: string
  category: HabitCategory
  frequencyType: HabitFrequencyType
  targetCount: number
  difficulty: HabitDifficulty
  attributeRewards: HabitAttributeRewardResponse[]
  isCompletedToday: boolean
  currentStreak: number
  longestStreak: number
}
